deployment/cloud_composer_template.tf

```diff
@@ -99,6 +99,14 @@ resource "google_project_iam_member" "monitoring_viewer_role" {
   role     = "roles/monitoring.viewer"
 }
 
+resource "google_project_iam_member" "compute_instance_admin_role" {
```

Comment:
Could you help have a quick try to have this role in default service account only? And see if this permission is needed here. Sorry I should mention it earlier.